app/assets/javascripts/discourse/app/templates/invites/show.hbs

```diff
@@ -19,71 +23,84 @@
         {{#unless isInviteLink}}
           <p>
             {{html-safe yourEmailMessage}}
-            {{#if externalAuthsEnabled}}
+            {{#unless externalAuthsOnly}}
               {{i18n "invites.social_login_available"}}
-            {{/if}}
+            {{/unless}}
           </p>
         {{/unless}}
 
-        <form>
-
-          {{#if isInviteLink}}
-            <div class="input email-input">
-              <label>{{i18n "user.email.title"}}</label>
-              {{input type="email" value=email id="new-account-email" name="email" autofocus="autofocus"}}
+        {{#if externalAuthsOnly}}
+          {{#if authOptions}}
+            {{#unless isInviteLink}}
```

Comment:
This is still a mess I need to clean up.

Reply:
Can't see a way to get around the mess. Part of it is because we mixed both invite links and non-invite links within the same model